modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1242,44 +1242,43 @@ def compute_part_size(indexer, remote_part, part_idx, axis):
 
     @classmethod
     @wait_computations_if_benchmark_mode
-    def binary_operation(cls, axis, left, func, right):
+    def binary_operation(cls, left, func, right):
         """
-        Apply a function that requires two PandasDataframe objects.
+        Apply a function that requires two ``PandasDataframe`` objects.
 
         Parameters
         ----------
-        axis : {0, 1}
-            The axis to apply the function over (0 - rows, 1 - columns).
         left : np.ndarray
-            The partitions of left PandasDataframe.
+            The partitions of left ``PandasDataframe``.
         func : callable
             The function to apply.
         right : np.ndarray
-            The partitions of right PandasDataframe.
+            The partitions of right ``PandasDataframe``.
 
         Returns
         -------
         np.ndarray
             A NumPy array with new partitions.
         """
-        if axis:
-            left_partitions = cls.row_partitions(left)
-            right_partitions = cls.row_partitions(right)
-        else:
-            left_partitions = cls.column_partitions(left)
-            right_partitions = cls.column_partitions(right)
-        func = cls.preprocess_func(func)
-        result = np.array(
+
+        def op_with_empty_check(x, y, *args, **kwargs):
```

Comment:
The algebra operators should be generic as possible. I think this corner case should be handled in `_copartition`.

Reply:
Implementation of this corner case was changed in 9e90c5e9fca72c98182a13929a895d95cf22b1a9. For now, the implementation doesn't do a remote call in case the left or right operands are empty and just return an empty Modin DataFrame with correct index/columns as pandas does.

I don't see the reason to create an empty dataframes of right partitions in `_copartition` because it will just add superfluous execution time.

Reply:
Makes sense to me.

Reply:
Since #4059 seems to be resolved in this PR, please include it in the PR description and in release notes.

Reply:
#4059 was added in PR title/commit message/release notes